src/Control/Concurrent/CachedIO.hs

```diff
@@ -26,6 +27,11 @@ import Control.Monad.Catch (MonadCatch, onException)
 import Control.Monad.IO.Class (liftIO, MonadIO)
 import Data.Time.Clock (NominalDiffTime, addUTCTime, getCurrentTime, UTCTime)
 
+-- | A cached IO action in some monad @m@. Use 'runCached' to extract the action when you want to query it.
+--
+-- This newtype is intended to make it harder to accidentally 'Control.Monad.join' (and defeat all caching) when the fetch action and the outer monad are the same.
```

Comment:
Documentation that compares the current design with a forgotten old design tends to be confusing after a while. Could you remove this bit?

Reply:
Replacing `harder` with `difficult` would make this description stand alone, I think.

Reply:
Difficult compared with what? This sentence serves only to warn haskell-cached-io developers away from removing the newtype, which I don't think they will do. It certainly does not belong in the haddock.

Reply:
No, it is meant to warn library users away from immediately unwrapping the newtype constructor. If it is not doing that, it must be rewritten.

Reply:
I can put something like:

> Note that using 'Control.Monad.join' when the cached action and the outer monad are the same will ignore caching

Otherwise i'm inclined to remove this line altogether

Reply:
I normally hate it when libraries don't include these footguns in their documentation, so I decided to keep the above

Reply:
I understand now. I think my problem is that it takes contortions to actually use `join` on this function. It may be better to say something like

````
Note: if you create a `Cached m` inside a monad `m`, it can be tempting to run the action immediately; but doing this forfeits the cache.
````
or
````
Footgun warning: the following code builds a @Cache@ but forfeits caching behaviour:

  fetchFromDatabase :: IO Foo
  
  f :: IO Foo
  f = cachedIO 5 fetchFromDatabase >>= runCache
````